.github/workflows/linting.yml

```diff
@@ -22,4 +22,4 @@ jobs:
 
       - name: Run pylint
         run: |
-          pylint daliuge-common daliuge-translator daliuge-engine
+          pylint daliuge-common daliuge-translator daliuge-engine --fail-under=9 --fail-on=E                    
```

Comment:
These settings mean Pylint will return an error code of 0 unless we are either under 9 or fail on Errors. 
We should be at a 10.0 currently, and should have no Errors after this PR. What these settings also allow is for us to enable Warnings (thus seeing them introduced in the code), without us Failing any tests. This means we can, if we want to, enable Warnings at some point whilst we address them incrementally, without it affecting the CI. Something to think about moving forward I expect. 

We score a 9.47(?) with the Warnings enabled, so it's unlikely we'd ever go below 9. It is necessary to get the `--fail-on=E` to work properly, however.